CMS-Enterprise/easi-app
# [EASI-4628] Add priority category

# EASI-4628

[Figma](https://www.figma.com/design/iGatgTnlzTuqW9CQFYHDam/TRB-Intake-e2e?node-id=6924-14527&node-type=CANVAS&t=xVTUUBapCJpGmA7O-0)

## Description
- Add priority category fields to TRB guidance letter form "guidance and insights" step
- Display guidance letter insights by category
- Update admin notes to display category when selecting an insight as the guidance letter section ([Figma](https://www.figma.com/design/iGatgTnlzTuqW9CQFYHDam/TRB-Intake-e2e?node-id=6934-21194&t=9XqtxNZg106bzJ1V-1))
- Renamed recommendations to insights throughout frontend codebase
- Updated redirect so that previous "recommendations" links redirect to new "insights" route
- Unit tests

**Uncategorized Insights:**
- Previously created insights now have the "uncategorized" category ([Figma](https://www.figma.com/design/iGatgTnlzTuqW9CQFYHDam/TRB-Intake-e2e?node-id=7066-20844&t=fNz9mLji86tyKdcr-1))
- This category cannot be selected from form
- When these insights are updated, user must select new category

> [!IMPORTANT]  
> Success messages are not displaying correctly after adding or updating an insight. This bug existed before these changes and will be fixed in a separate PR.


## How to test this change
- Seed mock data before testing
- TRB request titled "Case 8 - Guidance letter in review with document" has guidance letter with both categorized and uncategorized guidance
- Guidance letter form
  - Guidance and Insights form category field should work as expected
  - Test reordering insights
    - Insights can only be ordered within category
    - If insight is updated to new category, it moves to the end of that category list
- Admin notes
  - Add new note
  - Select "Guidance letter" for "What is this note about?"
  - Insight labels should show category to match Figma
- Public guidance letter should display categories
  - /trb/guidance-letter/:trbRequestId

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
